massa-consensus-exports/src/block_status.rs

```diff
@@ -58,6 +63,45 @@ impl From<&BlockStatus> for BlockStatusId {
     }
 }
 
+/// A structure defining whether we keep a full block with operations in storage
+/// or just the raw signed block without operations
+#[derive(Debug, Clone)]
+pub enum StorageOrBlock {
+    /// Keep a full storage with operations
+    Storage(Storage),
+    /// Keep only the block header and list of ops (but not the ops)
+    Block(Box<SecureShareBlock>),
+}
+
+impl StorageOrBlock {
+    /// Return a clone of the underlying block
+    pub fn clone_block(&self, block_id: &BlockId) -> SecureShareBlock {
+        match self {
+            StorageOrBlock::Storage(storage) => storage
+                .read_blocks()
+                .get(block_id)
+                .expect("block absent from its own storage")
+                .clone(),
+            StorageOrBlock::Block(block) => *block.clone(),
+        }
+    }
+
+    /// Convert any StorageOrBlock variant into a StorageOrBlock::Block variant.
+    /// This effectively drops the operations of the block.
+    pub fn strip_to_block(&mut self, block_id: &BlockId) {
```

Comment:
maybe rename to into_variant_block ?

Reply:
it does not have the typical "into" behavior that consumes `self` and produces a new `Self`, here it mutates `&mut self`